social/twitter/27-twitter.js

```diff
@@ -11,77 +11,126 @@ module.exports = function(RED) {
     var localUserCache = {};
     var userObjectCache = {};
     var userSreenNameToIdCache = {};
+   
+    RED.nodes.registerType("twitter-credentials", TwitterCredentialsNode, {
+        credentials: {
+            consumer_key: { type: "password" },
+            consumer_secret: { type: "password" },
+            access_token: { type: "password" },
+            access_token_secret: { type: "password" },
+            access_token_bearer: { type: "password" }
+        }
+    });
 
     function TwitterCredentialsNode(n) {
-        RED.nodes.createNode(this,n);
+        RED.nodes.createNode(this, n);
         this.screen_name = n.screen_name;
         if (this.screen_name && this.screen_name[0] === "@") {
             this.screen_name = this.screen_name.substring(1);
         }
         if (this.credentials.consumer_key &&
             this.credentials.consumer_secret &&
             this.credentials.access_token &&
-            this.credentials.access_token_secret) {
+            this.credentials.access_token_secret &&
+            this.credentials.access_token_bearer) {
 
             this.oauth = {
                 consumer_key: this.credentials.consumer_key,
                 consumer_secret: this.credentials.consumer_secret,
                 token: this.credentials.access_token,
-                token_secret: this.credentials.access_token_secret
+                token_secret: this.credentials.access_token_secret,
+                token_bearer: this.credentials.access_token_bearer
             }
             this.credHash = crypto.createHash('sha1').update(
-                this.credentials.consumer_key+this.credentials.consumer_secret+
-                this.credentials.access_token+this.credentials.access_token_secret
+                this.credentials.consumer_key + this.credentials.consumer_secret +
+                this.credentials.access_token + this.credentials.access_token_secret
+                + this.credentials.access_token_bearer
             ).digest('base64');
             var self = this;
-            if (localUserCache.hasOwnProperty(self.credHash)) {
-                this.localIdentityPromise = Promise.resolve(localUserCache[self.credHash]);
-            } else {
-                this.localIdentityPromise = this.get("https://api.twitter.com/1.1/account/settings.json").then(function(body) {
-                    if (body.status === 200) {
-                        localUserCache[self.credHash] = body.body.screen_name;
-                        self.screen_name = body.body.screen_name;
-                    } else {
-                        self.warn("Failed to get user profile");
+
+            const needle = require('needle');
```

Comment:
I see here you are using `needle` however `needle` is not a dependency in this package (`request` is a dependency).

I am not against using `needle` (in fact it is a reasonable alternative considering `request` is depreciated) BUT for me to accept this PR, I would suggest the rest of the HTTP requests are changed to use `needle` as well. 

If you can do that - great. 

PS, you would also need to add a dependency in the package.json for `needle`